src/sagemaker/session.py

```diff
@@ -1286,6 +1301,12 @@ def process(
         inferred_resources_config = update_nested_dictionary_with_values_from_config(
             resources, PROCESSING_JOB_PROCESSING_RESOURCES_PATH, sagemaker_session=self
         )
+        environment = resolve_value_from_config(
+            direct_input=environment,
+            config_path=PROCESSING_JOB_ENVIRONMENT_PATH,
+            default_value={},
```

Comment:
same comment

Also, environment is a required param to this method today. We should probably switch it to optional and throw an error if the user didnt provide anything and the config didnt either. (Can look at what we did for IAM Roles when adding default configs)

Reply:
As per Processing API it is not a required parameter. It could be a miss in SDK.